### PR TITLE
Updated httest formula to 2.4.23

### PR DESCRIPTION
**Pull request by lwe**
Including enabled modules for JavaScript (via spidermonkey), HTTP2
(via nghttp2) and XML/HTML (via system libxml2).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by lwe** [comment]
Seems to fail to build on Sierra and El Capitan, with:

```
ld: file not found: /usr/lib/system/libsystem_darwin.dylib for architecture x86_64 # Sierra
ld: file not found: /usr/lib/system/libsystem_symptoms.dylib for architecture x86_64 # El Capitan
```

Root cause seems a problem with the system libxml2 provided by Xcode.

**Comment by ilovezfs** [comment]
ENV["SDKROOT"] = MacOS.sdk_path

File: Formula/httest.rb [hunk 1]
```diff
@@ -16,17 +16,28 @@ class Httest < Formula
...
+  depends_on "nghttp2"
+  depends_on "spidermonkey"
 
   def install
     # Fix "fatal error: 'pcre/pcre.h' file not found"
     # Reported 9 Mar 2017 https://sourceforge.net/p/htt/tickets/4/
     (buildpath/"brew_include").install_symlink Formula["pcre"].opt_include => "pcre"
     ENV.prepend "CPPFLAGS", "-I#{buildpath}/brew_include"
 
+    # Fix "ld: file not found: /usr/lib/system/libsystem_darwin.dylib" for lxml
```
Comment (ilovezfs) [inline Formula/httest.rb]: lxml isn't relevant for this one is it?

Comment (lwe) [reply Formula/httest.rb]: true it's libxml2.

File: Formula/httest.rb [hunk 2]
```diff
@@ -16,17 +16,28 @@ class Httest < Formula
...
+    ENV["SDKROOT"] = MacOS.sdk_path if MacOS.version == :sierra || MacOS.version == :el_capitan
```
Comment (lwe) [inline Formula/httest.rb]: @ilovezfs, would it make more sense to write as `if MacOS.version < :high_sierra`?

Comment (ilovezfs) [reply Formula/httest.rb]: nope this is fine

Comment (ilovezfs) [reply Formula/httest.rb]: you should use block-if though to keep the line length down

**Comment by lwe** [comment]
👍  thanks for the fast review, adapted accordingly.

File: Formula/httest.rb [hunk 3]
```diff
@@ -16,17 +16,30 @@ class Httest < Formula
...
     system "./configure", "--disable-dependency-tracking",
                           "--prefix=#{prefix}",
+                          "--enable-lua-module",
+                          "--enable-html-module",
+                          "--enable-xml-module",
+                          "--enable-h2-module",
+                          "--enable-js-module",
                           "--with-apr=#{Formula["apr"].opt_bin}",
-                          "--enable-lua-module"
+                          "--with-nghttp2=#{Formula["nghttp2"].opt_bin}",
```
Comment (ilovezfs) [inline Formula/httest.rb]: does it not find this and spidermonkey on its own automatically?

Comment (ilovezfs) [reply Formula/httest.rb]: also are you sure you mean opt_bin not opt_prefix?

Comment (lwe) [reply Formula/httest.rb]: sure, will try without explicit references.

**Comment by lwe** [comment]
Updated. Also marked lua, spidermonkey and nghttp2 as _recommended_ dependencies, as `httest` really works fine without these.

**Comment by ilovezfs** [comment]
Thanks for another contribution to Homebrew @lwe!

**Pull request closed without merging**